### ISSUE TITLE
[Nightly] "Delete" in search field - UI glitch or on purpose?

### ISSUE THREAD

**Issue by MagpieFourtyTwo** [cgeo/cgeo#16173]
### Describe your problem!

In c:geo's home screen enter the search field (by tapping the magnifying glass) - the term "Delete" appears in the search field.
Why?  Forgotten debug output?  Or do I just don't get it ...?  ;)

### How to reproduce?

.

### Actual result after these steps?

.

### Expected result after these steps?

.

### Reproducible

Yes

### c:geo Version

2024.10.05-NB-33b5ab3

**Comment by moving-bits** [comment]
Can't reproduce currently. This is how it looks like for me in emulator with current master after having tapped the search icon on home screen:

![image](https://github.com/user-attachments/assets/b9549262-c6a6-47f2-960e-7170172f1ec0)

Is it just the exact string "Delete" for you, or something more? (Then we can try to re-trace for that string var.)

**Comment by MagpieFourtyTwo** [comment]
Over here: 
![Screenshot_20241012_212919_cgeo_geocaching_MainActivity.jpg](https://github.com/user-attachments/assets/3b39f271-38e1-4e7e-a17f-b311819d6e6a)

**Comment by moving-bits** [comment]
This is pretty strange. On a German-based c:geo it says "Caches/Trackables suchen".

**Comment by MagpieFourtyTwo** [comment]
Ok ... over here it's in "Default language", on an English based Android. So probably a translation resp. NLS dictionary problem ...?

**Comment by moving-bits** [comment]
Ok, it get's even stranger: I've switched my (physical) test device to system language "english" and restarted c:geo. I've two installations on that device, one with many caches (current release) - that one shows "Search caches/trackables" -, and an "empty" one (developer build) - that one shows "Search..." in that very same place.

**Comment by MagpieFourtyTwo** [comment]
;)

Overloaded translations?
Don't know how NLS is implemented in c:geo, but if it works with string IDs, maybe an ID is used multiple times ...? 

Although the dots in "Search..." can even be the result of "not enough space for 'caches/trackables'" ...?

**Comment by MagpieFourtyTwo** [comment]
Meanwhile it showed some even more strange texts, but currently it says "Search caches/trackables" - at least over here.  Looks a bit like a compilation problem.

As for the issue:  Although it can possibly be considered "solved" (by the work of nice elves), I would still leave it open for a while, to have a place where we can quickly post (and track) if it changes again.  Cause in fact I'm not really convinced that the real cause is fixed ... and if there is e. g. a problem with NLS, this may occur in other places, too.

**Comment by MagpieFourtyTwo** [comment]
And with today's nightly I get the text "Bike" ... 

![Screenshot_20250110_090056_cgeo_geocaching_MainActivity_copy_512x112.jpg](https://github.com/user-attachments/assets/b403a553-1b22-4da4-88a2-7f2cff91ad2d)

**Comment by ztNFny** [comment]
"Search..." is what happens when the field is too short to show "Search caches/trackables" - I'll add an invisible space after the "/" to enable breaking there.

I'm unable to reproduce the problem though. @MagpieFourtyTwo using some framework like Magisk, Xposed, .. that could mess with this? The string is a "perfectly normal" one, nothing special about it:
    <item
        android:id="@+id/menu_gosearch"
        android:icon="@drawable/ic_menu_search"
        android:title="@string/search_bar_hint"

**Comment by MagpieFourtyTwo** [comment]
No, nothing like that, it's just Android 15 with Honor's MagicOS 9.0 right out of the box with no OS-relevant modifications at all. At least not intentionally ... ;)

**Comment by ztNFny** [comment]
unless someone else comes along I'd say it's most likely that something in the OS is messing this up. The IDs are correct in the apk.

**Comment by MagpieFourtyTwo** [comment]
Absolutely fine for me. Just wanted to point it out - not that there might be problems with NLS after all ...
While this is BTW the only "strange" string I noticed so far, so it has prio Z or so, anyway.  ;)

**Comment by MagpieFourtyTwo** [comment]
My very personal Easter egg got a new color today ... while yesterday's Nightly finally gave me a "Search caches/trackables", today it reads "Found Places". ;)

Have to check on next event if I'm really the one and only lucky guy (still guess there are others, but they just didn't realize yet, esp. as many users do not even know the benefits of this search field, most of them use the magnifier in the bottom menu, even for GC codes).  And perhaps we can even discover something in common ...

**Comment by murggel** [comment]
> Ok, it get's even stranger: I've switched my (physical) test device to system language "english" and restarted c:geo. I've two installations on that device, one with many caches (current release) - that one shows "Search caches/trackables" -, and an "empty" one (developer build) - that one shows "Search..." in that very same place.

 _posted by @moving-bits on [Oct 13, 2024](https://github.com/cgeo/cgeo/issues/16173#issuecomment-2409066110)_

Same here:
If I start a developer build (even in landscape mode), only "Search..." or "Suchen..." is displayed, while with the current nightly version (apk) "Search caches/ trackables..." is displayed.

Also, opening a cache from the search list does not work with the developer build.
@moving-bits  Does the "Search..."-version works for you?

**Comment by MagpieFourtyTwo** [comment]
I over here have seen the wildest strings in there ... best so far was "Backup / Restore".  :)
But right now - much to my surprise - it really shows "Search".  But with no dots at all.  Although there's plenty of room, about three times the length of the word "Search" is simply empty.  🤷🏼

But whatever string is shown, the search itself never had a single problem (AFAIR).
